logger.c

```diff
@@ -0,0 +1,47 @@
+#include "logger.h"
+
+static logger_pair_t logger[] = {
+  {
+    .log_name  = "server",
+    .log_value = LOG_KIND_SERVER,
+    .length    = sizeof("server") - 1
+  },
+  {
+    .log_name  = "thermocouple",
+    .log_value = LOG_KIND_THERMOCOUPLE,
+    .length    = sizeof("thermocouple") - 1
+  }
+};
+
+static const size_t
+LOGGER_OPTIONS_SIZE = sizeof(logger) / sizeof(logger_pair_t);
+
+static bool
+compare_str(const char* buf, const int buflen, const char* buf2, const int buf2len)
+{
+  if(buflen != buf2len) return false;
+
+  for(int i = 0; i < buflen; i++)
+    if(buf[i] != buf2[i]) return false;
+
+
+  return true;
+}
+
+void
+set_log(const char* buf, const unsigned set, uint8_t* log_bits)
+{
+  const int buflen = strlen(buf);
+
+  for(int i = 0, opt = 0; i < LOGGER_OPTIONS_SIZE; i++, opt++)
+  {
+    if(!compare_str(buf, buflen, logger[i].log_name, logger[i].length)) continue;
+
+    bool is_set = *log_bits & (0x1 << opt);
+
+    if(is_set == true && set == 0)  *log_bits -= logger[i].log_value;
+    if(is_set == false && set == 1) *log_bits += logger[i].log_value;
+    break;
```

Comment:
LMAO

![sumimasen](https://github.com/SecMeant/pico-furnace/assets/21361739/c35c7a60-6f4d-4f0d-9aba-e439299dcce8)

this is how boolean logic works:
```c
unsigned bits;

bits |= 1u << 2; // set bit at position 2
bits &= ~(1u << 3);  // clear bit at position 3
```